HW06.Rmd

```diff
@@ -48,45 +73,49 @@ ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")
 ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
   #these are removed instead of averaged because they simply do not correlate particularly well. 
 ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
-
 ```
 
-
-
 Do the t-test and make a table of the t-test results!
 
 ```{r t.test for difference between a bunch of genes}
-#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
 detach(package:EnsDb.Hsapiens.v75, unload = T)
 detach(package:ensembldb, unload = T)
 
-#add in your own gene of interest!!! 
-genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
-
+### Added in genes HIF1 and HIF2 of interest
+genes_of_interest <- c("HIF1", "HIF2", "IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
 
 GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
 
 #Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
 
 #print out the t-test results
-```
-
 
+### Having some trouble getting my code for the T tests to work, have temporarily moved my code for organizing the code down to the graphing section so that the graphing section will run. Will look into this again when I get the chance!
+```
 
 Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
 
 ```{r create a graphing function}
-#to work in the tidyverse, it will be easier to make tidy the dataframe first
-
-
-
-#create a graphing function
-
-#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+GBM.reorganized <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c("group", "order"), convert = TRUE)
+
+graph <- function(x) 
+  {
+  gene_function <- GBM.reorganized  %>%
+    filter(gene_symbol == gene)

```

Comment:
Your codes here are similar to mine when initiating the t-test. I decided to reorganize the file first and filter gene before doing a for loop to run a t-test on each gene and that helped me. Maybe this can be something for you to look into. The only issue with my way of doing is that somehow my "gene" in for (gene in genes_of_interest) gets assigned as a variable and I think that led to why my second for loop is not working. :( Still trying to figure this part out!

Reply:
Thanks for the comments Nabilah! I actually had a similar idea yesterday to reorganize the file and filter but hadn't had time to work on it yet. I looked through your code and I'll definitely try something similar. If I figure out why your second loop isn't working or I run into a similar problem I'll give you a shout!